website/src/docs/hotchocolate/fetching-data/resolvers.md

```diff
@@ -0,0 +1,508 @@
+---
+title: "Resolvers"
+---
+
+import { ExampleTabs } from "../../../components/mdx/example-tabs"
+
+When it comes to fetching data in a GraphQL server, it will always come down to a resolver.
+
+**A resolver is a generic function that fetches data from an arbitrary data source for a particular field.**
+
+We can think of each field in our query as a method of the previous type which returns the next type.
+
+## Resolver Tree
+
+A resolver tree is a projection of a GraphQL operation that is prepared for execution.
+
+For better understanding, let's imagine we have a simple GraphQL query like the following, where we select some fields of the currently logged-in user.
+
+```graphql
+query {
+  me {
+    name
+    company {
+      id
+      name
+    }
+  }
+}
+```
+
+In Hot Chocolate, this query results in the following resolver tree.
+
+```mermaid
+graph LR
+  A(query: QueryType) --> B(me: UserType)
+  B --> C(name: StringType)
+  B --> D(company: CompanyType)
+  D --> E(id: IdType)
+  D --> F(name: StringType)
+```
+
+This tree will be traversed by the execution engine, starting with one or more root resolvers. In the above example the `me` field represents the only root resolver.
+
+Field resolvers that are subselections of a field, can only be executed after a value has been resolved for their _parent_ field. In the case of the above example this means that the `name` and `company` resolvers can only run, after the `me` resolver has finished. Resolvers of field subselections can and will be executed in parallel.
+
+**Because of this it is important that resolvers, with the exception of top level mutation field resolvers, do not contain side-effects, since their execution order may vary.**
+
+The execution of a request finishes, once each resolver of the selected fields has produced a result.
+
+_This is of course an oversimplification that differs from the actual implementation._
+
+# Defining a Resolver
+
+Resolvers can be defined in a way that should feel very familiar to C# developers, especially in the Annotation-based approach.
+
+## Properties
+
+Hot Chocolate automatically converts properties with a public get accessor to a resolver that simply returns its value.
+
+Properties are also covered in detail by the [object type documentation](/docs/hotchocolate/defining-a-schema/object-types).
+
+## Regular Resolver
+
+A regular resolver is just a simple method, which returns a value.
```

Comment:
A question that's often asked is hwo the naming convention works, e.g. Get prefix and Async suffix being dropped off, then lowercase of first letter = field name.

Should we add an aside/note here explaining that?

Reply:
I added this to the existing naming section for object types.